tests/src/core/testqgsexpression.cpp

```diff
@@ -3891,6 +3891,26 @@ class TestQgsExpression: public QObject
       QCOMPARE( QgsExpression::replaceExpressionText( input, &context ), expected );
     }
 
+    void testConcatNULLAttributeValue()
+    {
+      // Test that null integer values comming from provider are not transformed as 0
+      // https://github.com/qgis/QGIS/issues/36112
+
+      QgsFields fields;
+      fields.append( QgsField( QStringLiteral( "foo" ), QVariant::Int ) );
+
+      QgsFeature f;
+      f.initAttributes( 1 );
+      f.setAttribute( 0, QVariant( QVariant::Int ) );
```

Comment:
can you also test `QVariant(  )` ? (it should give the same result)

Reply:
Yes, done